org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/CompilationUnit.java

```diff
@@ -452,6 +454,20 @@ public IJavaElement[] codeSelect(int offset, int length, WorkingCopyOwner workin
 		return super.codeSelect(this, offset, length, workingCopyOwner);
 	}
 }
+private static ASTNode getParentNodeToUse(ASTNode node) {
+	if (node instanceof Name) {
+		ASTNode parentFinder = node.getParent();
+		while (parentFinder.getParent() != null && parentFinder.getParent() instanceof Type) {
+			parentFinder = parentFinder.getParent();
+		}
+		if (parentFinder.getParent() != null && parentFinder.getParent() instanceof ClassInstanceCreation) {
+			return parentFinder.getParent();
+		}
+	}
+	return node;
+}
```

Comment:
There is a trick here: imagine one type `new SomeParentType.SomeNestedType((SomeRandomOtherType)null, (YetAnotherRandomOtherType)null)`, then with the change, all the types would then lead to their constructor definition, while in practice, we'd only want it if selecting the `SomeNestedType`. So additional checks are necessary.